src/image_intel.cpp

```diff
@@ -20,29 +20,73 @@
 constexpr size_t IMAGE_A_ADDR = 0x20080000;
 constexpr size_t IMAGE_B_ADDR = 0x22480000;
 
+enum class BootPart
+{
+    image_a,
+    image_b,
+    unknown
+};
+
 /**
- * @brief Get current boot address
+ * @brief Get current boot partition
  */
-static size_t getBootAddress()
+static BootPart getBootPart()
 {
+    BootPart activePart = BootPart::unknown;
     const std::string bootcmdPrefix = "bootcmd=bootm";
+    std::string bootpart;
 
+    try
+    {
+        bootpart = exec(FW_PRINTENV_CMD " -n bootside 2>/dev/null");
+    }
+    catch (...)
+    {}
+    if (bootpart == "a\n")
+    {
+        return BootPart::image_a;
+    }
+    if (bootpart == "b\n")
```

Comment:
It seems to me the comparison with  `bootside=b` looks more reliable. In that case, `-n` is excess in `fw_setnev` call.
Also I would make some rearrange code to something like:
```cpp
try
{
    auto bootpart = exec(FW_PRINTENV_CMD " bootside 2>/dev/null");
    return (bootpart == "bootside=b") ? BootPart::image_b : BootPart::image_a;
}
catch (...)
{}
```

Reply:
And `catch` statement also requires a comment.
For example: `There is no such argument and it isn't an error.`

Reply:
done